### PR TITLE
austin.intermediate.summer2.wada.harry

### PR DESCRIPTION
**Pull request by harrywada**

### CODE REVIEW

File: homework/01LessonOne.js [hunk 1]
```diff
@@ -17,27 +17,27 @@ var myUndefined;
 //
 // What is the difference between null and undefined?
```
Comment (tjefferson08) [inline homework/01LessonOne.js:18]: Sneaky question here: add a comment with an answer for full points!

File: homework/01LessonOne.js [hunk 2]
```diff
@@ -146,54 +146,54 @@ var notNotUndefined;
 // Your Answer Goes Here:
 //
 // ****
-
+// truthiness is determined by the state of a boolean which depends on the state of the arguments provided, strings with content are considered true
```
Comment (tjefferson08) [inline homework/01LessonOne.js]: What about falsey values?

File: homework/01LessonOne.js [hunk 3]
```diff
@@ -210,8 +210,7 @@ var stringFourStrictNotEqualsNumberFour;
 // Your Answer Goes Here:
 //
 // ****
-
-
+// '==' roughly compares values, '===' must have the values exactly the same
```
Comment (tjefferson08) [inline homework/01LessonOne.js]: > roughly compares values

What does that mean?

File: apps/01PigLatin.js [hunk 4]
```diff
@@ -7,11 +7,80 @@ prompt.start();
...
-    // Your code here
+  // Concept Check
+  //
+  // Please explain why we are using String.prototype.toLowerCase() here.
+  // For more info:
+  // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/toLowerCase
+  //
+  // Your Answer:
+  // in order to ensure that there are no weird capitalizations in the middle of the word and that all words have consistent capitalization, String.prototype.toLowerCase() is used make all letters lowercase
+  word  = word.toLowerCase();
```
Comment (paulrenenichols) [inline apps/01PigLatin.js:18]: Good answer!

catbug

File: apps/01PigLatin.js [hunk 5]
```diff
@@ -7,11 +7,80 @@ prompt.start();
...
 
+  // Concept Check
+  //
+  // What is the significance of setting vowelIndex to -1?
+  // This might help:
+  // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/lastIndexOf
+  //
+  // Your Answer:
+  // vowelIndex is set to -1 to show that there is currently no value assigned to the variable, meaning that no vowels have been found
+  var vowelIndex = -1; // Set it to assume there are no vowels
```
Comment (paulrenenichols) [inline apps/01PigLatin.js:28]: Not quite. We are initializing **vowelIndex** to -1, so it does have a value assigned to it.

The value -1 indicates that we have not yet found a vowel.

File: apps/01PigLatin.js [hunk 6]
```diff
@@ -7,11 +7,80 @@ prompt.start();
...
+  }
+
+  // Concept Check
+  //
+  // Why are we using String.prototype.slice() here? What role does vowelIndex play?
+  // This might help:
+  // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/slice
+  //
+  // Your Answer:
+  // vowelIndex is the location of the vowel in the word, so String.prototype.slice() is used to select the part of the word before and after the index of the word
```
Comment (paulrenenichols) [inline apps/01PigLatin.js:61]: Perhaps you meant `before and after the index of the vowel`

File: apps/01PigLatin.js [hunk 7]
```diff
@@ -7,11 +7,80 @@ prompt.start();
...
+// Concept Check
+//
+// Make sure you run this app with node in the console before answering this question.
+// Also, read Lesson 1 in the textbook.
+// http://intermediate.austincodingacademy.com/_book/lesson_one.html
+//
+// What does prompt.get() do for us?
+//
+// Your Answer:
+// prompt.get() requests input from the user and inserts them into an array that can be accessed in the callback function
```
Comment (paulrenenichols) [inline apps/01PigLatin.js:83]: Great answer.

File: apps/02RockPaperScissors.js [hunk 8]
```diff
@@ -9,6 +9,43 @@ function rockPaperScissors(hand1, hand2) {
 
     // Write code here
 
+    //normalize input
+    hand1 = hand1.toLowerCase().trim();
+    hand2 = hand2.toLowerCase().trim();
+
+    //"else if" not necessary because if condition is satisfied, a return statement is always executed before continuing
+    //ensure input is valid
+    if((hand1 !== "rock" && hand1 !== "paper" && hand1 !== "scissors") ||
```
Comment (tjefferson08) [inline apps/02RockPaperScissors.js:18]: NOICE

File: apps/02RockPaperScissors.js [hunk 9]
```diff
@@ -9,6 +9,43 @@ function rockPaperScissors(hand1, hand2) {
...
+       (hand2 !== "rock" && hand2 !== "paper" && hand2 !== "scissors")){
+      return "invalid input";
+    }
+
+    //check for tie
+    if(hand1 === hand2){
+      return "It's a tie!";
+    }
+
+    //check if hand1 is rock
```
Comment (tjefferson08) [inline apps/02RockPaperScissors.js:28]: Dude. Awesome comments

File: homework/02LessonTwo.js [hunk 10]
```diff
@@ -26,6 +26,7 @@ var alwaysTrue = function () {
 // argument1 is equal to argument2
 
 function equals(argument1, argument2) {
+  return argument1 == argument2;
```
Comment (tjefferson08) [inline homework/02LessonTwo.js:29]: Nice

File: homework/02LessonTwo.js [hunk 11]
```diff
@@ -34,16 +35,23 @@ function equals(argument1, argument2) {
...
+function add(number1, number2){
+  return number1 + number2;
+}
 
 // Problem 4:
 // write a function named addThree(number1, number2, number3)
 // this function will add three numbers
 // you must call your function add() in addThree()
+function addThree(number1, number2, number3){
+  return add(add(number1, number2), number3);
```
Comment (tjefferson08) [inline homework/02LessonTwo.js:53]: Sweet

File: homework/02LessonTwo.js [hunk 12]
```diff
@@ -54,22 +62,22 @@ function lessThanOrEqualTo(parameter1, parameter2) {
 // What is a function? How do you define a function in Javascript?
 //
 // Your Answer Goes Here:
-//
+// a function is a method that has 0 or more inputs and executes commands, returning 0 or more outputs, to define a function, the structure is "function function_name(parameters){}"
```
Comment (tjefferson08) [inline homework/02LessonTwo.js:65]: short. sweet. me likey

File: homework/02LessonTwo.js [hunk 13]
```diff
@@ -54,22 +62,22 @@ function lessThanOrEqualTo(parameter1, parameter2) {
...
 //
 // Your Answer Goes Here:
-//
+// function function_name(parameters){}
 //
 // What is a parameter? What is an argument? Is there a difference between the two?
 //
 // Your Answer Goes Here:
-//
+// a parameter is the name of the variable when first defined in the function, arguments are the actual input that is used when the function is actually called
```
Comment (tjefferson08) [inline homework/02LessonTwo.js:80]: vocab FTW

File: homework/02LessonTwo.js [hunk 14]
```diff
@@ -108,6 +118,16 @@ function isDivisibleByThree(number) {
 // should return false if character is anything else
 
 function whichSpecies(character) {
+  switch(character){
+    case "scooby":
```
Comment (tjefferson08) [inline homework/02LessonTwo.js:122]: Nice! Why don't you need `break` statements here?

File: homework/03LessonThree.js [hunk 15]
```diff
@@ -20,16 +20,15 @@ var firstName = 'susan';
...
 
 // Problem 2:
 // What is the value of lastName at this point in the file?
-var valueOfLastName;
+var valueOfLastName = "plaid";
 
 // Problem 3:
 // Fix the following function so that it returns the first argument
 function returnTheFirstArgument(firstArgument) {
-  var firstArgument = 'bob';
```
Comment (tjefferson08) [inline homework/03LessonThree.js:32]: Sweet.

File: homework/03LessonThree.js [hunk 16]
```diff
@@ -41,7 +40,7 @@ function returnTheFirstArgument(firstArgument) {
 // What is scope?
 //
 // Your Answer Goes Here:
-//
+// Scope is the context in which a variable or function can be used. For instance, a variable created in a function can only be used in that function, its scope does not include anything beyond that function
```
Comment (tjefferson08) [inline homework/03LessonThree.js:43]: Nicely put

File: homework/03LessonThree.js [hunk 17]
```diff
@@ -79,7 +78,7 @@ var problemSevenAnswer;
 // What is the ternary operator?  How does it differ from normal conditional statements?
 //
 // Your Answer Goes Here:
-//
+// The ternary operator allows you to evaluate a boolean statement and provide two values, one to be returned if the statement is true, and one to be returned if the statement is false. This is also much easier than using "if" statements because it only takes one line.
```
Comment (tjefferson08) [inline homework/03LessonThree.js:81]: NIIICE. 

> evaluate a boolean statement

I'd rather you say boolean _expression_ here

File: homework/03LessonThree.js [hunk 18]
```diff
@@ -89,17 +88,19 @@ var problemSevenAnswer;
...
 // Create an array with 5 elements in it
-var lengthFiveArray;
+var lengthFiveArray = [1, 2, 3, 4, 5];
 
 // DO NOT MODIFY
 var nameArray = ['bob', 'fred', 'susan'];
 
 // Problem 10:
 // Replace the value 'fred' in nameArray with 'george'
+var index = nameArray.indexOf("fred");
```
Comment (tjefferson08) [inline homework/03LessonThree.js:102]: nice helper variable!

File: homework/03LessonThree.js [hunk 19]
```diff
@@ -110,10 +111,14 @@ var threeByThreeArray = [
 
 // Problem 11:
 // Replace the center element of threeByThreeArray, which is 5, with something else.
+threeByThreeArray[1][1] = 6;
 
 // Problem 12:
 // create a 2 by 2 (2 rows, 2 columns) nested array
-var twoByTwoArray;
+var twoByTwoArray = [
+  [1, 2],
```
Comment (tjefferson08) [inline homework/03LessonThree.js:119]: Boring! :P

**Pull request closed without merging**